### ISSUE TITLE
Download Speeds Slow Down After Initial 95% Completion, Possible DNS Query Issue with level3.blizzard.com

### ISSUE THREAD

**Issue by lonelywanderer** [tpill90/battlenet-lancache-prefill#126]
I'm experiencing an issue with the program where it downloads approximately 95% of the content at normal speed, but the download speed drastically slows down once it reaches that point. The download speed becomes extremely slow, making the remaining 5% take a long time to finish.

Additionally, I noticed that within a few hours, the program sends around 40,000 DNS queries to level3.blizzard.com. This might be related to the slowdown, and I'm wondering if the high volume of DNS queries could be causing or contributing to the issue.

**Comment by tpill90** [comment]
What game are you seeing this with?  Was the game previously prefilled? How fast is your internet?

**Comment by lonelywanderer** [comment]
I prefill multiple games and experience the issue at every one.

I prefill D2, D3, D4, Hearthstone, HotS, SC2, WC3: Reforged, WoW and WoW classic.

I did not prefill them before.

I have a 50k connection and it slows down to 10 kbit/s sometimes.

**Comment by tpill90** [comment]
To clarify, that's a 50kbit/s connection?

**Comment by lonelywanderer** [comment]
No, it's 50 mbit/s. I have an avarage download speed of 5-7 MB/s.

Sorry for the misunderstanding. Here in germany we call it x thousand ('k') kbit/s.

**Comment by dav3yblaz3** [comment]
i love blizzard and ai and robots please get this to test thank you team
